### PR TITLE
Fix apq8074 support

### PR DESCRIPTION
**Pull request by lumag**

### CODE REVIEW

**Comment by MarijnS95** [comment]
r-b on [`pil-squasher: write ELF header after reading it`](https://github.com/andersson/pil-squasher/pull/5/commits/645667a0f804418a5823c50d39c6bb1f72d1d54d), was already curious to see this missing in #4 but the PR was already merged :)

**Comment by lumag** [comment]
@MarijnS95 usually the ELF header is written when processing the phdr 0. However I stumbled upon a case which has the following headers:

```
$ readelf -hl /tmp/mba.mbn 
ELF Header:
  Magic:   7f 45 4c 46 01 01 01 00 00 00 00 00 00 00 00 00 
  Class:                             ELF32
  Data:                              2's complement, little endian
  Version:                           1 (current)
  OS/ABI:                            UNIX - System V
  ABI Version:                       0
  Type:                              EXEC (Executable file)
  Machine:                           QUALCOMM DSP6 Processor
  Version:                           0x1
  Entry point address:               0xd100000
  Start of program headers:          52 (bytes into file)
  Start of section headers:          0 (bytes into file)
  Flags:                             0x3
  Size of this header:               52 (bytes)
  Size of program headers:           32 (bytes)
  Number of program headers:         1
  Size of section headers:           0 (bytes)
  Number of section headers:         0
  Section header string table index: 0

Program Headers:
  Type           Offset   VirtAddr   PhysAddr   FileSiz MemSiz  Flg Align
  LOAD           0x001000 0x0d100000 0x0d100000 0x38278 0x38278 RWE 0
```

**Comment by MarijnS95** [comment]
Fwiw I vaguely recall seeing more than just the header in `mdt` files that was not referenced by any segment. Could have been garbage, or not :sweat_smile:, would have to check up on my local library. There was also a special "hash" type IIRC that may have to remain in the `mdt` file?

**Comment by lumag** [comment]
@MarijnS95 yes, the `mdt` file consists of ELF header + signature data. For most of the typical files these are segments 0 and 1 concatenated.

**Comment by MarijnS95** [comment]
@lumag yup, I thought `readelf` would render the magic flag for that though on program header 1:

```console
$ readelf -hl a530_zap.elf
ELF Header:
  Magic:   7f 45 4c 46 01 01 01 00 00 00 00 00 00 00 00 00
  Class:                             ELF32
  Data:                              2's complement, little endian
  Version:                           1 (current)
  OS/ABI:                            UNIX - System V
  ABI Version:                       0
  Type:                              EXEC (Executable file)
  Machine:                           QUALCOMM DSP6 Processor
  Version:                           0x1
  Entry point address:               0x5000
  Start of program headers:          52 (bytes into file)
  Start of section headers:          0 (bytes into file)
  Flags:                             0x3
  Size of this header:               52 (bytes)
  Size of program headers:           32 (bytes)
  Number of program headers:         3
  Size of section headers:           0 (bytes)
  Number of section headers:         0
  Section header string table index: 0

Program Headers:
  Type           Offset   VirtAddr   PhysAddr   FileSiz MemSiz  Flg Align
  NULL           0x000000 0x00000000 0x00000000 0x00094 0x00000     0
  NULL           0x001000 0x00007000 0x00007000 0x01988 0x02000     0x1000
  LOAD           0x003000 0x00005000 0x00005000 0x01324 0x01324 RWE 0x100000
```

**Comment by lumag** [comment]
No, unfortunately readelf skips that flag.

**Comment by MarijnS95** [comment]
But it is quite strange that your file does not show it... Oh well, as long as you can load it when only copying the ELF header and keeping the rest of the program headers in `.bxx` files. The only alternative is copying anything up until the first program header (typically `0x1000`, IIRC the signature was duplicated here too).

**Comment by lumag** [comment]
Unlike the rest of the apq8074 that I have here, mba.mdt was not
signed at all. Rest of the firmware files contain the hash table
(without certificate chain and actual signature). This one doesn't
even has a hash table.

**Comment by lumag** [comment]
@andersson gracious ping

File: pil-splitter.c [hunk 1]
```diff
@@ -53,6 +53,7 @@ int main(int argc, char **argv)
 	size_t phoff;
 	bool is_64bit;
 	size_t offset;
+	size_t hashoffset;
```
Comment (MarijnS95) [inline pil-splitter.c]: You added this local variable and assign to it, but I don't see it being read anywhere?

**Review by MarijnS95 — APPROVED** [review]
R-b after dropping the unused `hashoffset` local variable.

File: pil-splitter.c (hunk 1 above)
Comment (lumag) [reply pil-splitter.c]: @MarijnS95 done

**Pull request merged**